### ISSUE TITLE
Reactivate Twitter account 

### ISSUE THREAD

**Issue by navarroaxel** [tldr-pages/tldr#5898]
We should reactivate the official twitter account for `tldr-pages`: https://twitter.com/tldr_pages 

Comments from @waldyrious in [`Gitter`](https://gitter.im/tldr-pages/tldr?at=608aaf81da3e853b33ddf219):

> (1) it would still be associated to his personal email address and phone number (security measures), so we need a solution to that (for example, so that we don't depend on him in case we need to recover access to account or there's a security issue like logging in from an unusual location or device); and (2) it's not about me getting the password, since that just moves the problem to another place. What we need is a scalable way to have shared access to the account without sharing its credentials. I believe that's what TweetDeck allows, but I haven't explored that in detail yet to be sure.

Possible solution: TweetDeck and the `Twitter together` GitHub action (https://github.com/gr2m/twitter-together).

**Comment by nicokosi** [comment]
I am curious: what will this account be used for? New command notifications? Team changes? Other news?

**Comment by sbrl** [comment]
We could have tweets about:

 - New pages that have been merged
 - GitHub releases (which often contain important information for client authors)

**Comment by CleanMachine1** [comment]
Any updates for this.

Could also add things like general questions for TLDR features which are being discussed, such as our current conversation for having a sending off of data about peoples tldr commands. EG missing pages

**Comment by sbrl** [comment]
The most immediate question though is who has the password for it.

Twitter together sounds like a good way to do things, so I suggest we set that up first in a new GitHub repo in the tldr-pages org and then look at developing some tooling to automatically commit new tweets.

**Comment by CleanMachine1** [comment]
I'd imagine the creator of TLDR

**Comment by bl-ue** [comment]
@ rprieto made tldr, but I believe @ igorshubovych made it.

**Comment by bl-ue** [comment]
A novel idea: we could have a **private** https://github.com/tldr-pages/secrets repo, that's only visible to org _owners_ (**not** even members), who have a long time of trust.  We could store info such as passwords, keys, etc. in it.  Also stuff like @tldr-bot's credentials.

**Comment by CleanMachine1** [comment]
I agree. Org owners since others don't need that secret anyway

**Comment by CleanMachine1** [comment]
Or just tell the password to the people on safe messaging services. Why store it in a repo, Github has been hacked before im pretty sure

**Comment by bl-ue** [comment]
Storing it in a repository would be its history, so that we don't have to share them with new org owners all over again.
There's nothing new about that idea — https://github.com/nodejs/secrets & https://github.com/nodejs/keys exist but are private.

**Comment by sbrl** [comment]
What about cost. Does GitHub have any plans to charge for private repos again as they did in the past?

**Comment by bl-ue** [comment]
No idea, but I suspect not — why would they take it off and then put it back on?  If they ever did, we could always do something like transfer it to your or my account or any member of our team that has a pro account, and add the org owners as collaborators 🤔

**Comment by SethFalco** [comment]
I don't think it's a good idea to store it in a GitHub repository. 🤔 
Uploading data somewhere is the same has giving that entity access to the data.

In other words, you aren't just creating a GitHub repository to share your passwords.
You're giving both GitHub and Microsoft the password to your Twitter account.

There is no reason to do this regardless of context, large business, community project, or even an individual.
GitHub is _not_ a password manager.

If you're keen on sharing the password, I'd strongly recommend setting up an end-to-end encrypted solution at the very least.

In my opinion, the best would just be to have a free organization on Bitwarden for tldr pages manages by the 2 most trusted members, then use Bitwarden Send (free end-to-end encrypted sharing) to share the password with individuals that have any reason to access the Twitter account.

Being a trusted member of tldr doesn't mean you have reason to access the Twitter account. We should be following the [principle of the least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). The only reason to ever see the password is if you are both trusted, and will actually do something with it.

> only visible to org owners (not even members)

That's still too many people.

> Or just tell the password to the people on safe messaging services. Why store it in a repo, Github has been hacked before im pretty sure

Strongly agree with this concern.

**Comment by sbrl** [comment]
Great idea about Bitwarden there. Is there a cost to bitwarden, and do they have a program for open-source orgs to get it for free?

**Comment by SethFalco** [comment]
A Bitwarden organization for 2 people is free. (Bitwarden Send can be used to send passwords to other members securely.)

I don't think they have anything specific for open-source specifically, but for non-profits they offer a 25% discount.

If open-source is included in that definition, I'm not sure. I can vouch the process doesn't include checking if you're a registered charity/non-profit in any jurisdiction, so I'm guessing it's case-by-case.

If this doesn't suit you, we can always explore other password managers. I just suggested Bitwarden because it's my preferred password manager, plus it's open-source.

**Comment by sbrl** [comment]
Hrm, I'm not convinced that 2 people is really useful there @SethFalco, given that one could simply just share the password for example. We'd want to be able to support maybe ~3 to ~5 people. Still, it wouldn't hurt to drop them an email to ask?

**Comment by SethFalco** [comment]
Understood. In that case, it doesn't hurt to ask.
Who's gonna drop a message? I know back in the Weblate issue an email was formally agreed upon by the members, would you want to do that again? Or is an informal message fine?

Easiest way to connect privately would probably be: https://bitwarden.com/contact/

**Comment by CleanMachine1** [comment]
Should this really be discussed openly?

**Comment by SethFalco** [comment]
> one could simply just share the password

I would say Bitwarden is significantly more useful if you intend to share the password over GitHub, Gitter, email, or some other insecure means that is not end-to-end encrypted.

I agree with your point only if you're referring to sharing it via and end-to-end encrypted [Matrix](https://matrix.org/) room, end-to-end encrypted email like [Tutanota](https://tutanota.com/), or some other end-to-end encrypted means.

**Comment by CleanMachine1** [comment]
Could we not just use GPG encryption then just send files through something simple

**Comment by sbrl** [comment]
Yeah, we could draft an email here, and then I can send it off.

I do agree that Bitwarden has benefits, but given we're an open-source org with [limited funds](https://liberapay.com/tldr-pages/income/), I'm not sure we can afford it right now (@ ostera's share is for our domain tldr.sh, and mine is for the hosting of the tldr-bot).

-------

Hello,

I'm contacting you representing tldr-pages (<https://github.com/tldr-pages/tldr>) to ask about whether you offer any discounts or free tiers for Bitwarden for open-source organisations? We are considering using Bitwarden to securely store a number of secrets that we have accumulated.

Best regards,

-------

I feel like I'm a bit too concise here.

**Comment by SethFalco** [comment]
> I feel like I'm a bit too concise here.

I'd say concise is fine. They don't _need_ any of the fluff, they know what open-source is, and are familiar with GitHub so can easily see the notability of tldr, etc.

Unless it'll be one of those things where Sales gets the email, but doesn't know anything about the industry they're in Sales for. ^-^'

If you want to be safe, some things you could consider is:
* Make note of the number of stargazers of the main repo. (33.4k)
* Make note that it's a community project solely run by volunteers, and has no ties to a corporation.
* Consider proposing it as a corporate sponsor. (With link/logo at bottom of README.)

I realize the sponsor one may be controversial and will require a wider discussion with other maintainers.
Since tldr is trying to get free services outside what's readily available, it doesn't sound unreasonable to return the favor.

Some repositories on GitHub have a section for non-monetary corporate sponsors, for example [JetBrains Licenses for Open Source Development](https://www.jetbrains.com/community/opensource/#support):
* [FlaUI](https://github.com/FlaUI/FlaUI#acknowledgements) - Acknowledgements

Some other notable examples, but not on GitHub READMEs:
* [Inkscape](https://inkscape.org/support-us/sponsors/)
* [openSuse](https://en.opensuse.org/Sponsors)

---

Note: The bullet point listed above consists of **ideas to consider only**, and do not indicate any preference for inclusion.

**Comment by CleanMachine1** [comment]
Considering that we are asking for free stuff, the best you can do is show some gratitude. 

I personally wouldn't mind a advert type of thing, if they would be inclined.

**Comment by sbrl** [comment]
Noting down our infrastructure doesn't seem like such a bad idea - either at the bottom of the README as a list or elsewhere in our documentation is probably a good place. An advertisement is different in my opinion, and not acceptable. Simply noting briefly that they generously provide us with a service would be ok though - /cc @owenvoke, @mebeim, and maybe @agnivade on that one.

Stargazers / solely by volunteers is a good idea to mention too.

**Comment by CleanMachine1** [comment]
By advert, I meant a mention in the README. Like you have suggested

**Comment by CleanMachine1** [comment]
Has anything been done for this, unless I am stupid (probably the answer) whats wrong with public key encryption via gpg, since most of us have GPG

clearly visable from github.com/{{username}}.gpg

**Comment by sbrl** [comment]
Getting back to the core issue at hand here - reactivating the twitter account - I think given the age of this issue it's probably a good idea if we go ahead and create a brand new account. Any thoughts for the name? The account in question we can't access is `@tldr_pages`.

**Comment by marchersimon** [comment]
Would Twitter allow `@tldr-pages`? Other alternatives would be `@tldrpages`, `@tldr.pages` or maybe even `@tldr-project`.

**Comment by SethFalco** [comment]
Oh! I was unaware that no one had access to the previous account, that's unfortunate. 🤔 

Twitter usernames:
* May only contain letters, numbers, and underscores.
* Must be between 4 and 15 characters. (inclusive)

Looks like `@tldrpages` would be a winner.

**Comment by marchersimon** [comment]
If the new account gets big enough we could maybe also ask Twitter to remove the old one and rename the new one to `@tldr_pages`.

**Comment by sbrl** [comment]
I could create the new account, and privately message active org owners on Gitter with the login credentials - encrypting with gpg where available.

See my comment here about content: <https://github.com/tldr-pages/tldr/issues/5898#issuecomment-840903959>

**Comment by navarroaxel** [comment]
> I could create the new account, and privately message active org owners on Gitter with the login credentials - encrypting with gpg where available.

This is not needed @sbrl. I talked with @waldyrious and he will send you the credentials of the official account.

**Comment by CleanMachine1** [comment]
What about tldr.sh?

**Comment by sbrl** [comment]
> This is not needed @sbrl. I talked with @waldyrious and he will send you the credentials of the official account.

Oh awesome @navarroaxel! That works too. I await the credentials then. I assume via Gitter or maybe email?

**Comment by marchersimon** [comment]
Any updates yet?

**Comment by sbrl** [comment]
Heya, @marchersimon! Thanks for reminding me to update this issue. I now have access to the account through TweetDeck, and I think I can add other people to TweetDeck too. I can't create an API key though to set  up Twitter Together though I don't think through TweetDeck, unless I just haven't found the option to do so.

tldr-pages org owners, if you can confirm to me your twitter account handle I will add you via TweetDeck.

**Comment by navarroaxel** [comment]
🦆 [@navarroaxel](https://twitter.com/navarroaxel)

**Comment by sbrl** [comment]
Update: I now have access to the twitter account itself. Along with @ waldyrious we're now working on setting up <https://github.com/gr2m/twitter-together/>. It looks like you need to apply for a developer account, so we're working through that.

Other things that came to mind in our email conversation (private via email to exchange passwords etc):

> Also a matter for later discussion: among the people with TweetDeck access, there should be some sort of agreement (ideally something explicitly documented) about how to handle the account: how to handle notifications (e.g. if someone sees them, do they disappear for the others?), what sort of replies and mentions are appropriate, etc.

--@ waldyrious

@ waldyrious also updated the profile banner / avatar to match our latest branding

**Comment by sbrl** [comment]
Phew! That took a while. Scheduled a tweet every other day showcasing a bunch of different clients using Tweet Deck

I tried to pick those which have been most recently updated.

**Comment by sau226** [comment]
Came across this discussion on GH today while looking at collaboration ideas for another OSS org, and just wanted to share an idea based off what our org does: [1Password ](https://1password.com/) offers their Teams plan free for open source projects - the instructions on how to get it are in the [1Password/1password-teams-open-source](https://github.com/1Password/1password-teams-open-source) GitHub repo.

Different projects' philosophies/approaches may not allow for the use of 1Password, but if yours does, its worth giving things a go. They went and built out an [item sharing](https://blog.1password.com/psst-item-sharing/) tool recently, and the requirements for the free account aren't too onerous (free licenses for all core contributors, 2 year license that needs to be renewed for free via email).

**Comment by CleanMachine1** [comment]
https://twitter.com/tldr_pages

This has been done. Closing!

**Comment by sbrl** [comment]
We have yet to set up twitter together though, but I suppose that can be another issue.
